execution_engine/src/core/engine_state/upgrade.rs

```diff
@@ -328,4 +342,70 @@ where
 
         Ok(())
     }
+
+    pub(crate) fn ensure_consistent_contract_state(
```

Comment:
I think the name would be better as `create_accumulation_purse_if_required`.  Also, a doc comment would be helpful.

Reply:
Did this